tests/test_estimators/test_lowess.py

```diff
@@ -15,3 +18,15 @@ def test_obvious_usecase():
     y = np.ones(x.shape)
     y_pred = LowessRegression().fit(X, y).predict(X)
     assert np.isclose(y, y_pred).all()
+
+
+def test_custom_error_for_zero_division():
+    x = np.arange(0, 100)
+    X = x.reshape(-1, 1)
+    y = np.ones(x.shape)
+    estimator = LowessRegression(sigma=1e-10).fit(X, y)
+
+    with pytest.raises(
+        ValueError, match=re.escape("Weights, resulting from `np.exp(-(distances**2) / self.sigma)`, are all zero.")
+    ):
+        estimator.predict(X[:10] + 0.5)
```

Comment:
An offset is required, otherwise `X` to predict is same as `X` in `fit` method, leading to weight of 1 for the corresponding value.

Reply:
Maybe add that in the codebase as a comment?